nixos/modules/services/networking/wgautomesh.nix

```diff
@@ -0,0 +1,140 @@
+{ lib, config, pkgs, ... }:
+  let cfg = config.services.wgautomesh;
+in
+  with lib;
+  with builtins;
+{
+  options.services.wgautomesh = {
+    enable = mkEnableOption (mdDoc "Enable the wgautomesh daemon");
+    logLevel = mkOption {
+      type = types.enum [ "trace" "debug" "info" "warn" "error" ];
+      default = "info";
+      description = mdDoc "wgautomesh log level (trace/debug/info/warn/error)";
+    };
+    interface = mkOption {
+      type = types.str;
+      description = mdDoc "Wireguard interface to manage (it is NOT created by wgautomesh, you should use another NixOS directive to create it such as `networking.wireguard.interfaces.wg0 = {...};`)";
+      example = "wg0";
+    };
+    gossipPort = mkOption {
+      type = types.port;
+      description = mdDoc "wgautomesh gossip port, this MUST be the same number on all nodes in the wgautomesh network";
+      default = 1666;
+    };
+    enableGossipEncryption = mkOption {
+      type = types.bool;
+      default = true;
+      description = mdDoc "Enable encryption of gossip traffic. If this is set, a common secret key must be copied at /var/lib/wgautomesh/gossip_secret on all nodes.";
```

Comment:
`/var/lib/wgautomesh/gossip_secret` is used multiple times, what about assigning it to a variable or a module option?